### PR TITLE
Added new sync modes and order sync from date

### PR DESCRIPTION
**Pull request by jnmrclmbsjse**
This PR includes:
1. Added new sync modes `Product only`, `No sync`.
image
2. Added ability to set order sync date from.
image
3. Disabled payments for modes above.
image

### CODE REVIEW

File: src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php [hunk 1]
```diff
@@ -33,6 +39,10 @@ public function updateWithIgnore($order_id): void
     {
         global $storekeeper_ignore_order_id;
 
+        if (!$this->isSyncAllowed($order_id)) {
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php:42]: manual sync still should be allowed though, how you made it now the button will say success but nothing will happen

File: src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/EventsHandler.php [hunk 2]
```diff
@@ -255,6 +270,106 @@ private function processFullSyncEvents($events)
...
+        if (!is_array($events)) {
+            return;
+        }
+
+        $taskData = ['storekeeper_id' => (int) $this->getId()];
+        $storekeeper_id = (int) $this->getId();
+        $categoryType = $this->getCategoryType($storekeeper_id) ?? $this->getCategoryTypeFromWordpress($storekeeper_id);
+        $module = $this->getModule();
+
+        foreach ($events as $id => $event) {
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/EventsHandler.php:284]: That's really a lot of duplicate code. split into functions and call them inside the `processXXXEvents`

File: src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/InfoHandler.php [hunk 3]
```diff
@@ -95,6 +96,19 @@ public static function getExtras()
         return $extras;
     }
 
+    public static function getActiveCapabilities(): array
+    {
+        $activeCapabilities = [];
+        if (
+            !EventsHandler::isEventsDisabled('payments') &&
+            'yes' === StoreKeeperOptions::get(StoreKeeperOptions::PAYMENT_GATEWAY_ACTIVATED, 'yes')
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/InfoHandler.php:104]:
```suggestion
           StoreKeeperOptions::isPaymentsEnabled()
```

File: src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php [hunk 4]
```diff
@@ -92,4 +103,23 @@ public function addShopProductIdsToMetaData($order, $data)
...
         $order->add_meta_data(self::SHOP_PRODUCT_ID_MAP, $id_map);
     }
+
+    protected function isSyncAllowed(int $orderId): bool
+    {
+        if (EventsHandler::isEventsDisabled('orders')) {
+            return false;
+        }
+
+        if (0 !== $orderId) {
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php:113]: what are the cases when `$orderId === 0` ?

File: src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php (hunk 1 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php:41]: This won't affect manual sync as `OrderHandler` only handles those woocommerce hooks.
I added a note on metabox when order sync is disabled.

File: src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php [hunk 5]
```diff
@@ -92,4 +103,23 @@ public function addShopProductIdsToMetaData($order, $data)
 
         $order->add_meta_data(self::SHOP_PRODUCT_ID_MAP, $id_map);
     }
+
+    protected function isSyncAllowed(int $orderId): bool
+    {
+        if (EventsHandler::isEventsDisabled('orders')) {
```
[the remark this answers is not in the document]
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php]: Moved functions to `StorekeeperOptions`.

File: src/StoreKeeper/WooCommerce/B2C/Core.php [hunk 6]
```diff
@@ -132,14 +133,20 @@ public function __construct()
 
         $this->setUpdateCheck();
         $this->setLocale();
-        $this->setOrderHooks();
-        $this->setCustomerHooks();
+        if (!EventsHandler::isEventsDisabled('orders')) {
```
[the remark this answers is not in the document]
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Core.php]: Changed into positive checking and moved functions to `StorekeeperOptions`.

File: src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/InfoHandler.php (hunk 3 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/InfoHandler.php]: Changed into positive checking and moved functions to `StorekeeperOptions`.

File: src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php (hunk 4 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Tools/OrderHandler.php]: Removed checking of order ID and changed logic to positive checking.

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/OrderSyncMetaBox.php [hunk 7]
```diff
@@ -56,6 +56,17 @@ final public function renderSyncBox($post): void
         }
 
         $submitLabel = esc_html__('Force sync', I18N::DOMAIN);
+
+        $manualSyncWarning = '';
+        if (!StoreKeeperOptions::isOrderSyncEnabled()) {
+            $manualSyncWarningText = __("All orders won't be synced automatically with your currently selected Synchronization mode, but you can do it manually using the Force sync below", I18N::DOMAIN);
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/OrderSyncMetaBox.php:62]: missing escaping

File: src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/EventsHandler.php [hunk 8]
```diff
@@ -382,33 +202,155 @@ private function processOrderOnlyEvents($events)
         foreach ($events as $id => $event) {
             $details = $event['details'] ?? [];
             $eventType = $event['event'];
-            switch ("$module::$eventType") {
+
+            $fullEventType = "$module::$eventType";
+
+            $this->handleOrderEvents($fullEventType, $taskData, $details);
+
+            switch ($fullEventType) {
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/EventsHandler.php:210]: maybe change the stock as well to function? than consistent with the rest

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/OrderSyncMetaBox.php (hunk 7 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/OrderSyncMetaBox.php]: Changed to `esc_html__`.

File: src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/EventsHandler.php (hunk 8 above)
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Endpoints/Webhooks/EventsHandler.php]: Moved to methdo `handleProductStockEvents`.

**Pull request merged**